Sources/Apodini/Semantic Model Builder/OpenAPI/OpenAPIInterfaceExporter.swift

```diff
@@ -3,44 +3,83 @@
 //
 
 @_implementationOnly import OpenAPIKit
-@_implementationOnly import struct Vapor.Abort
+@_implementationOnly import Vapor
 import Foundation
 
 class OpenAPIInterfaceExporter: StaticInterfaceExporter {
     static var parameterNamespace: [ParameterNamespace] = .individual
 
     let app: Application
     var documentBuilder: OpenAPIDocumentBuilder
-    let configuration: OpenAPIConfiguration
+    var configuration: OpenAPIConfiguration
 
     required init(_ app: Application) {
         self.app = app
-        self.configuration = OpenAPIConfiguration(from: app)
+        if let storage = app.storage.get(OpenAPIStorageKey.self) {
+            self.configuration = storage.configuration
+        } else {
+            self.configuration = OpenAPIConfiguration()
+        }
         self.documentBuilder = OpenAPIDocumentBuilder(
-            configuration: configuration
+            configuration: &configuration
         )
+        setApplicationServer(from: app)
+        updateStorage()
     }
 
     func export<H: Handler>(_ endpoint: Endpoint<H>) {
         documentBuilder.addEndpoint(endpoint)
+        
+        // set version information from APIContextKey, if version was not defined by developer
+        if self.configuration.version == nil {
+            self.configuration.version = endpoint.context.get(valueFor: APIVersionContextKey.self)?.description
+            updateStorage()
+        }
     }
 
     func finishedExporting(_ webService: WebServiceModel) {
         serveSpecification()
+        updateStorage()
+    }
+    
+    private func setApplicationServer(from app: Application) {
+        let isHttps = app.vapor.app.http.server.configuration.tlsConfiguration != nil
+        let host = app.vapor.app.http.server.configuration.hostname
+        let port = app.vapor.app.http.server.configuration.port
+        if let url = URL(string: "\(isHttps ? "https" : "http")://\(host):\(port)") {
+            self.configuration.serverUrls.insert(url)
+        }
```

Comment:
As far as I can see we could also use the HTTP configurations for this information that are defined in the Configuration DSL.

Reply:
included this in: 43128c6, in case there is no `HTTPConfiguration` provided, this will still fall back to using `app.vapor.app.http.server.configuration`